dev/devicelab/bin/run.dart

```diff
@@ -42,6 +42,12 @@ String luciBuilder;
 /// Whether to exit on first test failure.
 bool exitOnFirstTestFailure;
 
+/// Path for test results.
+///
+/// If passed along with a task name, the Cocoon test results will be stored in this file.
+/// If passed without a task name, the file contents will be uploaded to Cocoon.
+String resultsPath;
```

Comment:
Can we separate the functionality with commands? eg 

* run.dart test ...
* run.dart upload_metrics ...

That way we can extend this file later with a compile command to optimize the resources usage.

Reply:
I created a separate `bin/test_runner.dart` to reduce the complexity of the migration, and only added `upload-metrics` for now. Once we have migrated off of Cocoon agents, we can remove the global args code in `bin/run.dart`.

I can follow up in another PR with the `test` command.